apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -11,18 +11,20 @@
 from .options import SOMATileDBContext
 
 
-def build_index(
-    keys: np.typing.NDArray[np.int64], context: SOMATileDBContext
+def tiledbsoma_build_index(
+    keys: np.typing.NDArray[np.int64], context: SOMATileDBContext = None, thread_count: int = 4  # type: ignore
```

Comment:
A few things here:

1. the `type: ignore` here is hiding a real error: `context = None` is not a valid value, since None is not a SOMATileDBContext. Instead, it should be specified as an `Optional[SOMATileDBContext]`.
2. When you start getting into optional arguments, it’s best to use keyword-only arguments, since there is no logic to the order that things might be omitted in (e.g. `tiledbsoma_build_index(some_keys, None, 5)` makes no more or less sense than `tiledbsoma_build_index(some_keys, 5)`
3. (minor) So that users get better type annotation information on `tiledbsoma_build_index`, let’s have its return be annotated as `IndexLike` rather than our internal-only type.

So, overall, this should look like:

```python
from somacore.query.types import IndexLike

def tiledbsoma_build_index(
    keys: np.typing.NDArray[np.int64],
    *,
    context: Optional[SOMATileDBContext] = None,
    thread_count: int = 4,
) -> IndexLike:
```

Reply:
All sound good.

Reply:
Done